### DISCUSSION TITLE
AttributeError: module 'pandas._libs.missing' has no attribute 'NAType'

### DISCUSSION THREAD

**Discussion by JayBaywatch** [duckdb/duckdb#5616] [Q&A] [answered]
Today I started getting the following exception on import of duckdb in python.

AttributeError: module 'pandas._libs.missing' has no attribute 'NAType'

I've tried changing the numpy and pandas versions but neither seem to resolve the issue. Has something changed? I am on version 0.6.0 of duckdb, but the issue is also present for me in version 0.6.1

My base version of numpy is 1.17.2 which would ideally remain pinned because my version of pandas is 0.25.3.

Does anyone have any insight as to what may be causing this import exception?

**Comment by Mytherin** [comment, marked as answer]
We have recently added support for the Pandas `NAType` - which was likely introduced after `0.25.3`. We could try to detect that an old version of Pandas is being used and catch the error - but is there anything stopping you from just upgrading your Pandas version?

**Reply by JayBaywatch** [reply]
as a firm, we are still on pandas 0.25.3 because we have many, many internal packages that would require testing and updating, which we have not had the time to do. duckdb 0.5.1 still works so I guess I can just use that. We have noted in the past, that updating pandas can sometimes have minor differences in the answers some of our stuff produces and as a finance shop, that's always bad news.

**Reply by Tishj** [reply]
I can't seem to install pandas `0.25.3` so I can't verify, but looking at the code this should be fixed.
We recently switched to lazy loading the imports, so it should definitely not be problem on `import` anymore.
And on top of that, a recent PR I made adds checking for the attribute before importing it - so it shouldn't spit out an attribute error anymore if `NAType` is not present in this case.
